### PR TITLE
[docs] Pipeline loading

### PR DESCRIPTION
**Pull request by stevhliu**
This PR refactors the `loading.md` doc to focus solely on pipelines (since there is quite a bit to know about them now) and reorganizes it a bit to accommodate the new `from_pipe` API. Additional changes include:

- remove the "Overview" doc from the loading section in favor of slimming down the docs and relying on better doc titles to help users navigate
- remove the "Reuse components across pipelines" section entirely to avoid confusion and because its way easier with `from_pipe` now
- move the loading scheduler and model sections from `loading.md` to `schedulers.md` to create a new "Load schedulers and models" doc
- move the `from_pipe` example for community pipelines to the "Load community pipelines and components" doc

### CODE REVIEW

**Review by yiyixuxu — APPROVED** [review]
thanks!

File: docs/source/en/using-diffusers/custom_pipeline_overview.md [hunk 1]
```diff
@@ -18,15 +18,17 @@ specific language governing permissions and limitations under the License.
...
+There are two types of community pipelines, those stored on the Hugging Face Hub and those stored on Diffusers GitHub repository. Refer to this [table](./contribute_pipeline#share-your-pipeline) for a comparison of Hub vs GitHub community pipelines.
 
-<Tip warning={true}>
+<hfoptions id="community">
+<hfoption id="Hub pipelines">
 
-🔒 By loading a community pipeline from the Hugging Face Hub, you are trusting that the code you are loading is safe. Make sure to inspect the code online before loading and running it automatically!
+To load a Hugging Face Hub community pipeline, pass the repository id of the community pipeline to the `custom_pipeline` argument and the model repository where you'd like to load the pipeline weights and components from. For example, the example below loads a dummy pipeline from [hf-internal-testing/diffusers-dummy-pipeline](https://huggingface.co/hf-internal-testing/diffusers-dummy-pipeline/blob/main/pipeline.py) and the pipeline weights and components from [google/ddpm-cifar10-32](https://huggingface.co/google/ddpm-cifar10-32):
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/custom_pipeline_overview.md:28]: Would it make sense to briefly differentiate between Hub pipelines and the ones that reside in our repository? 

(Hub pipelines may have everything custom, such as the scheduler, model, pipeline code, etc. Repo custom pipelines are only allowed to have custom pipelining code.)

**Review by sayakpaul — APPROVED** [review]
Sweet

File: docs/source/en/using-diffusers/custom_pipeline_overview.md (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/custom_pipeline_overview.md:28]: Added a brief sentence about their differences and linked to this [table](https://huggingface.co/docs/diffusers/main/en/using-diffusers/contribute_pipeline#share-your-pipeline) for more details.

**Pull request merged**